tests/unit/components/AppNavigation/AppNavigation.spec.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @copyright Copyright (c) 2020 Azul <azul@riseup.net>
+ *
+ * @author Azul <azul@riseup.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { mount } from '@vue/test-utils'
+import { emit } from '@nextcloud/event-bus'
+import AppNavigation from '../../../../src/components/AppNavigation/AppNavigation.vue'
+
+describe('AppNavigation.vue', () => {
+	'use strict'
+
+	describe('by default', () => {
+		it('is open', () => {
+			const wrapper = mount(AppNavigation)
+			expect(wrapper.vm.$data.open).toBe(true)
+		})
+	})
+	describe('toggle via event bus', () => {
+		it('toggle to turn it off', () => {
+			const wrapper = mount(AppNavigation)
+			emit('toggle-navigation', {open: undefined})
+			expect(wrapper.vm.$data.open).toBe(false)
+		})
+
+		xit('toggle without params', () => {
+			const wrapper = mount(AppNavigation)
+			emit('toggle-navigation')
+			expect(wrapper.vm.$data.open).toBe(false)
+		})
```

Comment:
Not sure if we want to support toggling the nav without any args.

Currently it is not working. I tried to implement it by adding a default to the bus handler but that did not help.

Reply:
I think we don't want to do that. Do you have a use-case in mind in which that would be beneficial?

Reply:
I don't know... I just noticed that in the toggle function toggling as in
```
open = !open
```
was implemented and i kept that.

```
emit('toggle-navigation', {open: undefined})
```
seems like an odd syntax for that to me.

If we don't want to support it at all then i would remove the check for undefined and treat `open: undefined` just like
```
emit('toggle-navigation', {open: false})
```

Reply:
Okay... I dropped the test for the syntax without `open` and kept the old behavior for `open: undefined` because there may be apps that rely on it.
Docs say that `open: false` will close. That's what this pr achieves.